pkg/router/gloo.go

```diff
@@ -276,6 +276,7 @@ func (gr *GlooRouter) getGlooUpstreamKubeService(canary *flaggerv1.Canary, svc *
 	if glooUpstreamWithConfig != nil {
 		configSpec := glooUpstreamWithConfig.Spec
 		upstreamSpec = gloov1.UpstreamSpec{
+			Labels:                      configSpec.Labels,
```

Comment:
glooUpstreamWithConfig.Metadata.Labels right?

Reply:
I don't think so? This defines from the configSpec what labels to pass to the other upstream objects. glooUpstreamWithConfig.Metadata.Labels might make sense here: https://github.com/KnechtionsCoding/flagger/blob/35c8957a5585f9aa7d4446fd211c0289baa7a6ea/pkg/router/gloo.go#L310 but I think that declaration is already correct. 

But I may be misunderstanding the flow here.